modules/light-clients/08-wasm/types/vm.go

```diff
@@ -114,7 +114,10 @@ func wasmInstantiate(ctx sdk.Context, cdc codec.BinaryCodec, clientStore storety
 	checksum := cs.Checksum
 	resp, err := instantiateContract(ctx, clientStore, checksum, encodedData)
 	if err != nil {
-		return errorsmod.Wrap(ErrWasmContractCallFailed, err.Error())
+		return errorsmod.Wrap(ErrVMError, err.Error())
+	}
+	if resp.Err != "" {
+		return errorsmod.Wrap(ErrWasmContractCallFailed, resp.Err)
 	}
 
 	if err = checkResponse(resp); err != nil {
```

Comment:
To restore the old behaviour, I'd pass `result.Ok` as the argument here and only handle the response internally